src/ic-cdk/src/api/stable.rs

```diff
@@ -161,6 +162,142 @@ impl io::Write for StableWriter {
     }
 }
 
+/// A writer to the stable memory which first writes data to a buffer and flushes the buffer to
+/// stable memory each time it becomes full. This reduces the number of system calls to
+/// `stable64_write` and `stable64_grow` which have relatively large overhead.
+struct BufferedStableWriter<M: StableMemory = CanisterStableMemory> {
+    /// The offset of the next write.
+    offset: u64,
```

Comment:
Nit: `stable_memory_offset_in_bytes` and `stable_memory_capacity_in_pages` would improve readability.

Reply:
No longer relevant since I'm now using `BufWriter` from `std`